framework/core/Project/Lang.pm

```diff
@@ -42,18 +42,65 @@ use Vcs::Git;
 
 our @ISA = qw(Project);
 my $PID  = "Lang";
+# TODO changeme
+my $RANDOM_TEST_FILE = "$SCRIPT_DIR/build-scripts/$PID/random_tests";
 
 sub new {
     my $class= shift;
+    my $work_dir = shift // "$SCRIPT_DIR/projects";
     my $name = "commons-lang";
     my $src  = "src/main/java";
     my $test = "src/test";
     my $vcs  = Vcs::Git->new($PID,
                              "$REPO_DIR/$name.git",
-                             "$SCRIPT_DIR/projects/$PID/commit-db",
+                             (shift // "$SCRIPT_DIR/projects/$PID/commit-db"),
```

Comment:
This subroutine now has two parameters. Can you make this more explicit by assigning them early on (`my (...) = @_`)? The call to `shift` at this location may cause trouble in the future.

Note that this comment also applies to the other project modules.